src/main/java/xyz/brassgoggledcoders/transport/predicate/NamePredicate.java

```diff
@@ -23,8 +25,19 @@ public boolean test(Entity entity) {
     public static NamePredicate create(PredicateParser predicateParser) throws PredicateParserException {
         if (predicateParser.hasNextString()) {
             return new NamePredicate(predicateParser.getNextString()::equals);
+        } else if (predicateParser.hasNextRegex()) {
+            String regex = predicateParser.getNextRegex();
+            return new NamePredicate(s -> {
+                try {
+                    return Pattern.matches(regex, s);
```

Comment:
I believe Pattern.matches compiles the Regex every time, it should probably be combined.